src/picongpu/include/particles/ionization/byCollision/ThomasFermi/ThomasFermi.def

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2016-2017 Marco Garten
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pmacc_types.hpp"
+
+namespace picongpu
+{
+namespace particles
+{
+namespace ionization
+{
+    /** Thomas-Fermi impact ionization model
+     *
+     * \tparam T_DestSpecies electron species to be created
+     * \tparam T_SrcSpecies particle species that is ionized
+     *         default is boost::mpl placeholder because specialization
+     *         cannot be known in list of particle species' flags
+     *         \see speciesDefinition.param
+     */
+    template<typename T_IonizationAlgorithm, typename T_DestSpecies, typename T_SrcSpecies = bmpl::_1>
+    struct ThomasFermi_Impl;
+
+    /** Thomas-Fermi impact ionization model
+     *
+     *
+     * This ionization model describes the atom inside the Thomas-Fermi framework
+     * in a self-consistent way. There the electrons are modeled as a density
+     * with respect to the distance from the core while the atomic core is often
```

Comment:
I think the documentation here is sufficient. If we later go to more detailed versions of the Thomas-Fermi model we can then document each variant on its own. For now it's safe to say that most Thomas-Fermi models assume that the atomic core has no spatial dimension and cannot interact except via the potential well it creates.